src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.Enumerated.cs

```diff
@@ -113,6 +114,7 @@ public static partial class AsnDecoder
         ///   <paramref name="expectedTag"/>.<see cref="Asn1Tag.TagValue"/> is not correct for
         ///   the method.
         /// </exception>
+        [RequiresDynamicCode("The enum type might not be available at runtime.")]
```

Comment:
Is that true?  I'd expect that a type passed as a generic wouldn't be trimmed, and that typeof(GenericParameter) would also be valid.

Reply:
I think the call to the non-generic `ReadEnumeratedValue` makes it unsafe. Its easy to fix by refactoring the code but wanted to get feedback on that since I don't own the code.

Reply:
Yes, the message does not describe the issue well. We should not need it in the first place.